### ISSUE TITLE
[Cart] Image link missing accessible name

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#590]
## Issue summary

The Product image link relies on the image `alt` text to provide an accessible name. Without an `alt` value, screen reader users will not be provided with context for its purpose.

- If the default `alt` text is set as the product title, screen reader users will hear the same information twice when navigating via link elements. This would lead to an overly verbose user experience.
- If the merchant includes descriptive `alt` text for the product image, this would not accurately convey the purpose of the link and may lead to a confusing user experience.

<details>
  <summary>Screenshots</summary>

  <img width="572" alt="Dawn cart page. Product image and title are highlighted." src="https://user-images.githubusercontent.com/1392632/132526696-dfe0bfbe-3454-4b1d-984e-dd01317a01a4.png">

</details>

#### HTML

```html
<a
  href="/products/alle-coffee-table-small-by-staffan-holm?variant=7111213908026"
>
  <img
    class="cart-item__image"
    src="…"
    alt=""
    loading="lazy"
    width="100"
    height="67"
  />
</a>
```

## Steps to reproduce

1. Go to a Dawn store with any supported browser
2. Add an item to the cart and proceed to the cart page
3. With an active screen reader, <kbd>Tab</kbd> to the product image link

### Expected

- Image to be skipped on <kbd>Tab</kbd>, moving to the product title link.

### Actual

- Image link is in Tab order without an accessible name.

## Recommendation

To avoid all the scenarios listed above:

1. Remove the `a` wrapping the `img` element.
2. Add the `onclick` attribute to the `img`, setting its value to `window.location.href='{product.url}'`.
3. Add `cursor: pointer;` to the `.cart-item__image` class definition.

With this change:

- Sighted mouse and touch users will still have the image click target to revisit the product page.
- Keyboard-only users will avoid an extra <kbd>Tab</kbd> by moving directly to the product title link.
- Screen readers will skip the image if no `alt` is set, or, consume the image description if one exists via virtual cursor.

#### HTML

```html
<img
  class="cart-item__image"
  src="…"
  alt=""
  loading="lazy"
  width="100"
  height="67"
  onclick="window.location.href='/products/crop-pants-olive?variant=39322207354902'"
/>
```

#### CSS

```css
.cart-item__image {
  /* … */
  cursor: pointer;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 2.0
- **Browser:** Any
- **Component:** Cart
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.4 Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
- **Severity:** High
- **Effort:** Medium

**Comment by bakura10** [comment]
@svinkle how does this compare with setting a tabindex to -1 on the image link?

**Comment by svinkle** [comment]
@bakura10 This removes the link from the natural tab order. Screen readers would still discover the link when using virtual/browse mode.
